tests/components/waterkotte_heatpump/test_config_flow.py

```diff
@@ -0,0 +1,158 @@
+"""Test the Waterkotte Heatpump config flow."""
+from unittest.mock import patch
+
+from pywaterkotte import AuthenticationException, ConnectionException, EcotouchTags
+
+from homeassistant import config_entries
+from homeassistant.components.waterkotte_heatpump.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+
+    with patch(
+        "pywaterkotte.ecotouch.Ecotouch.login",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.waterkotte_heatpump.async_setup_entry",
+        return_value=True,
+    ), patch(
+        "pywaterkotte.ecotouch.Ecotouch.read_value",
+        return_value=42,
+    ), patch(
+        "pywaterkotte.ecotouch.Ecotouch.decode_heatpump_series",
+        return_value="heatpump type",
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "heatpump type"
+    assert result2["data"] == {
+        "host": "1.1.1.1",
+        "username": "test-username",
+        "password": "test-password",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "pywaterkotte.ecotouch.Ecotouch.login",
```

Comment:
instead of patching it globally, you are supposed to patch it from within the component.
"homeassistant.components.waterkotte_heatpump. ..."